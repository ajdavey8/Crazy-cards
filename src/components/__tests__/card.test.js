import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Card from '../card';

const selectCard = jest.fn();

const card = {
  id: 1,
  cardType: 'Anywhere Card',
  apr: '33.9',
  balanceTransferDuration: 0,
  purchaseOfferDuration: 0,
  credit: 300
};

describe('Card', () => {
  const wrapper = shallow(<Card card={card} selectCard={selectCard} />);

  it('renders and matches snapshot', () => {
    const tree = renderer
      .create(
        <Card card={{ ...card, selected: true }} selectCard={selectCard} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an unselected card', () => {
    expect(wrapper.find('.card__heading').text()).toContain('Anywhere Card');
    expect(wrapper.find('.card__header-detail').text()).toContain(
      'Credit Available: £300'
    );
    expect(wrapper.find('.card__details').exists()).toBe(false);
  });

  it('renders a selected card', () => {
    wrapper.setProps({ card: { ...card, selected: true } });
    expect(wrapper.find('.card__heading').text()).toContain('Anywhere Card');
    expect(wrapper.find('.card__header-detail').text()).toContain(
      'Credit Available: £300'
    );
    expect(wrapper.find('.card__details').exists()).toBe(true);
  });
});
