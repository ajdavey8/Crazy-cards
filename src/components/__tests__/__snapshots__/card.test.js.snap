// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders and matches snapshot 1`] = `
<div
  className="card card--selected"
  onClick={[Function]}
>
  <div
    className="card__header"
  >
    <h3
      className="card__heading"
    >
      Anywhere Card
    </h3>
    <p
      className="card__header-detail"
    >
      Credit Available: Â£
      300
    </p>
  </div>
  <div
    className="card__details"
  >
    <p>
      Apr: 
      33.9
      %
    </p>
    <p>
      Balance Transfer Duration: 
      0
    </p>
    <p>
      Purchase Offer Duration: 
      0
    </p>
  </div>
</div>
`;
